docs/src/pipe.rst

```diff
@@ -106,3 +106,13 @@ API
     and call ``uv_accept(pipe, handle)``.
 
 .. seealso:: The :c:type:`uv_stream_t` API functions also apply.
+
+.. c:function:: int uv_pipe(uv_os_fd_t fds[2], int nonblock_read, int nonblock_write)
+
+    Create an pair of connected pipe handles. Data may be written to `fds[1]` and read from `fds[0]`.
+    The resulting handles can be passed to `uv_pipe_open`, used with `uv_spawn`, or for any other purpose.
```

Comment:
Is this is correct? I'm saying this because from what I can tell, uv_pipes are unix stream sockets and not pipes.

Reply:
Which part? This calls the `pipe2` or `pipe` syscall.

Reply:
The part where you say you can pass the handles to `uv_pipe_open`. I'm saying that because unix stream sockets are not the same as pipes and uv_pipes are implemented (on unices) using unix stream sockets

Reply:
`uv_pipe_bind` & friends use unix stream sockets, but `uv_guess_handle` expects that `uv_pipe_open` will also work with normal pipes. Thus, this code path has been getting tested any time that libuv have been put into a shell pipeline.

Reply:
Got it. Thanks for the explanation.